dll/win32/setupapi/devinst.c

```diff
@@ -2704,22 +2748,22 @@ HKEY WINAPI SetupDiCreateDeviceInterfaceRegKeyW(
                                             InfHandle,
                                             InfSectionName,
                                             SPINST_INIFILES | SPINST_REGISTRY | SPINST_INI2REG | SPINST_FILES | SPINST_BITREG | SPINST_REGSVR | SPINST_UNREGSVR | SPINST_PROFILEITEMS | SPINST_COPYINF,
-                                            hDevKey,
+                                            hDevParamKey,
                                             NULL,
                                             0,
                                             set->SelectedDevice->InstallParams.InstallMsgHandler,
                                             set->SelectedDevice->InstallParams.InstallMsgHandlerContext,
                                             INVALID_HANDLE_VALUE,
                                             NULL))
             {
-                RegCloseKey(hDevKey);
+                RegCloseKey(hDevParamKey);
                 return INVALID_HANDLE_VALUE;
             }
         }
     }
 
     SetLastError(rc);
-    return hDevKey;
+    return hDevParamKey;
```

Comment:
This one would be OK (it's only the `hDevParamKey` usage in the `SetupInstallFromInfSection` call above that I've still some doubts).